In [10]:
import numpy as np
import scipy.stats as stats

### Выбрать тест и проверить, есть  ли различия между выборками:
- 1 )  Даны две  независимые выборки. Не соблюдается условие нормальности
- x1  380,420, 290
- y1 140,360,200,900
### Сделайте вывод по результатам, полученным с помощью функции

In [11]:
selection_x1 = np.array([380,420, 290])
selection_y1 = np.array([140,360,200,900])

In [12]:
stats.mannwhitneyu(selection_x1, selection_y1)

MannwhitneyuResult(statistic=4.0, pvalue=0.29794154528258887)

так как pvalue=0.29, мы принимаем нулевую гипотезу, результат, различий нет

### Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?
- 1е измерение до приема препарата: 150, 160, 165, 145, 155
- 2е измерение через 10 минут: 140, 155, 150,  130, 135
- 3е измерение через 30 минут: 130, 130, 120, 130, 125

In [16]:
selection_1 = np.array([150, 160, 165, 145, 155])
selection_2 = np.array([140, 155, 150, 130, 135])
selection_3 = np.array([130, 130, 120, 130, 125])

In [17]:
stats.friedmanchisquare(selection_1, selection_2, selection_3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

так как pvalue=0.008, мы не принимаем нулевую гипотезу, так как значение ниже alpha=0.05, различия есть, препарат работает

### Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [18]:
stats.wilcoxon(selection_1, selection_2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

pvalue=0.0625, изменения не значительны, применяем нулевую гипотезу, препарат не работает

### Даны 3 группы  учеников плавания.
### В 1 группе время на дистанцию 50 м составляют:
- 56, 60, 62, 55, 71, 67, 59, 58, 64, 67
- Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
- Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [19]:
selection_1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
selection_2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
selection_3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

In [20]:
stats.kruskal(selection_1, selection_2, selection_3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

pvalue=0.065, изменения не значительны, применяем нулевую гипотезу, различий нет

### Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%
### 2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [23]:
party = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
mean = 2.5
alpha = 0.05

In [24]:
np.mean(party)

2.5279999999999996

In [25]:
t = (np.mean(party) - mean) / (np.std(party, ddof=1) / len(party) ** 0.5)
t

0.5630613661802959

по таблице распределения Стьюдента значение t при уровне значимости alpha = 0.05 и колличеством степеней свободы равным 9, мы имеем значение t=2.26 что